h/static/scripts/annotation-mapper.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+
+// Fetch the container object for the passed annotation from the threading
+// service, but only return it if it has an associated message.
+function getContainer(threading, annotation) {
+  var container = threading.idTable[annotation.id];
+  if (container === null || typeof container === 'undefined') {
```

Comment:
Right. But here I'm trying to preserve as near as possible the behaviour of the previous code. So I'd rather not change this in this case.